Src/BootCamp.Chapter/Credentials.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.ComponentModel;
+using System.IO;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    // TODO: make a struct and add validation and other needed methods (if needed)
+    public class Credentials
+    {
+        public readonly string Username;
+        public readonly string Password;
+
+        public Credentials(string username, string password)
+        {
+            if(String.IsNullOrWhiteSpace(username) || String.IsNullOrWhiteSpace(password))
+            {
+                throw new ArgumentException("Both Username and Password cannot be empty");
+            }
+            else
+            {
+                Username = username;
+                EncodedPassword tempPassword = new EncodedPassword(password);
+                Password = ByteToString(tempPassword.tempPassword);
+            }     
+        }
+
+        public readonly struct EncodedPassword
```

Comment:
Codacy found an issue: [Implement 'IEquatable<T>' in value type 'EncodedPassword'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5768165)